app/services/metrics/review_turnaround/per_user_project.rb

```diff
@@ -3,29 +3,57 @@ module ReviewTurnaround
     class PerUserProject < BaseService
       BATCH_SIZE = 500
 
+      def initialize(interval = nil)
+        @interval = interval
+      end
+
       def call
         process
       end
 
       private
 
       def process
-        filtered_reviews_ids.each_slice(BATCH_SIZE) do |batch|
-          Events::Review.eager_load(:review_request).find(batch).lazy.each do |review|
-            entity = find_user_project(review.owner, review.pull_request.project)
-            turnaround = calculate_turnaround(review)
+        ActiveRecord::Base.transaction do
+          filtered_reviews_ids.lazy.each_slice(BATCH_SIZE) do |batch|
+            Events::Review.joins(:project, :review_request, :pull_request, owner: :users_projects)
+                          .eager_load(:project, :review_request, owner: :users_projects)
```

Comment:
will this perform good enough when having a lot of data?

Reply:
Mmm I found it necessary because `joins` is being used as filter (INNER JOIN), and then `eager_load` to load in memory the necessary relations. Is there another way to make this?

Reply:
ok, let's go ahead and we can improve memory usage/performance later if needed